### PR TITLE
Instant Search: try using an overlay

### PR DESCRIPTION
**Pull request by bluefuton**
#### Changes proposed in this Pull Request:

Uses an overlay for search results and filtering, rather than hijacking the main page.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
It's an experimental feature and will merge into the `instant-search-master` branch.

#### Testing instructions:

To be added.

#### Proposed changelog entry for your changes:

No changelog required.

### CODE REVIEW

**Comment by bluefuton** [comment]
Todo: 
- [ ] add a way to close the overlay
- [ ] add the search box to the overlay
- [ ] remove search filters from the widget
- [ ] trigger the overlay from the search widget (on submit)
- [ ] trigger the overlay from a button in the main content - like the search icon in twentytwenty theme

File: modules/search/instant-search/components/overlay.scss [hunk 1]
```diff
@@ -0,0 +1,21 @@
+.jetpack-instant-search__overlay {
```
Comment (keoshi) [inline modules/search/instant-search/components/overlay.scss]: We'll probably need a ridiculous value for `z-index` to ensure the overlay is the top most layer on any site. I'd recommend starting with something like `9999999999999`. :)

Comment (jsnmoon) [reply modules/search/instant-search/components/overlay.scss]: Agreed :)

**Comment by jsnmoon** [comment]
@bluefuton I pushed out a substantial refactor with 1af7e93d1ce26c75e0ca6def5fcd9b0443b3d889, which adds focus event handlers to search inputs and significantly simplifies the SearchApp component; let me know what you think.

**Comment by bluefuton** [comment]
Simplification of `SearchApp` is great @jsnmoon! 💯

File: modules/search/class.jetpack-search.php [hunk 2]
```diff
@@ -207,6 +208,27 @@ public function init_hooks() {
 		add_action( 'jetpack_deactivate_module_search', array( $this, 'move_search_widgets_to_inactive' ) );
 	}
 
+	/**
```
Comment (bluefuton) [inline modules/search/class.jetpack-search.php]: What's the motivation for turning off the default search on the backend? Will it mean that users who aren't able to load Instant Search won't be able to use the basic search as a fallback?

Comment (jsnmoon) [reply modules/search/class.jetpack-search.php]: This was an attempt to address what to show behind the overlay when the user navigates directly to `/?s=seach-term-here`. In hindsight, I don't think this is a good idea; I'll revert this change today.

**Comment by bluefuton** [comment]
I think there are still a few things to figure out here, particularly:
- how should the overlay be triggered
- how to populate the search string back to the widget 

...but this is a solid start on the overlay. I say we ship this to `instant-search-master` and then iterate on it. ✅

File: modules/search/instant-search/components/overlay.jsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+const Overlay = ( { showOverlay, toggleOverlay, children } ) => {
+	const classNames = [ 'jetpack-instant-search__overlay' ];
+	if ( ! showOverlay ) {
+		classNames.push( 'is-hidden' );
+	}
+
+	return (
+		<div className={ classNames.join( ' ' ) }>
+			<button className="jetpack-instant-search__overlay-close" onClick={ toggleOverlay }>
+				{ __( 'Close' ) }
```
Comment (jsnmoon) [inline modules/search/instant-search/components/overlay.jsx]:
```suggestion
				{ __( 'Close', 'jetpack ) }
```

File: modules/search/instant-search/components/search-app.jsx [hunk 4]
```diff
@@ -43,40 +40,43 @@ class SearchApp extends Component {
...
-	prepareDomForMounting() {
-		// Clean up the page prior to mounting component
-		hideElements( this.props.themeOptions.elementSelectors );
-		document
-			.querySelectorAll( '.jetpack-instant-search-wrapper' )
-			.forEach( widget => removeChildren( widget ) );
 		document
-			.querySelectorAll( this.props.themeOptions.searchFormSelector )
-			.forEach( searchForm => removeChildren( searchForm ) );
+			.querySelectorAll( '.jetpack-instant-search-wrapper input.search-field' )
```
Comment (jsnmoon) [inline modules/search/instant-search/components/search-app.jsx]:
```suggestion
			.querySelectorAll( this.props.themeOptions.searchInputSelector )
```

I forgot to update this line when adding the searchInputSelector.

**Review by jsnmoon — APPROVED** [review]
Looks good overall; I found two enhancements we could make before shipping.

**Pull request merged**